hugo/content/docs/smodel.md

```diff
@@ -0,0 +1,554 @@
+
+*Properties*:
+
+* `layout: string` - *optional*: The layout of the compartment.
+* `opacity: number`: The opacity of the compartment. Defaults to `1`.
+
+*Inheritance:*
+
+[`SShapeElementImpl`](#sshapeelementimpl) &rarr; [`SChildElementImpl`](#schildelementimpl) &rarr; [`SParentElementImpl`](#sparentelementimpl) &rarr; [`SModelElementImpl`](#smodelelementimpl)
+
+*Default features:*
+
+* [boundsFeature](#boundsfeature)
+* [layoutContainerFeature](#layoutcontainerfeature)
+* [layoutableChildFeature](#layoutablechildfeature)
+* [fadeFeature](#fadefeature)
+
+### SGraphIndex
+
+A specialized model index that tracks outgoing and incoming edges.
+
+*Inheritance:*
+
+[`ModelIndexImpl`](#modelindeximpl)
+
+### ViewportRootElement
+
+Model root element that defines a viewport, so it transforms the coordinate system of its children with a scroll and zoom factor.
+
+*Properties:*
+
+* `scroll: Point`: The scroll offset of the viewport. Defaults to `{x: 0, y: 0}`.
+* `zoom: number`: The zoom factor of the viewport. Defaults to `1`.
+* `position: Point`: The position of the viewport. Defaults to `{x: 0, y: 0}`.
+* `size: Dimension`: The size of the viewport. Defaults to `{width: -1, height: -1}`.
+
+*Inheritance:*
+[`SModelRootImpl`](#smodelrootimpl) &rarr; [`SParentElementImpl`](#sparentelementimpl) &rarr; [`SModelElementImpl`](#smodelelementimpl)
+
+### SConnectableElementImpl
+
+A connectable element is on that can have outgoing and incoming edges. It can be the source or target element of an edge. There are two kinds of connectable elements: nodes and ports.
+
+*Properties:*
+
+* `strokeWidth: number`: The stroke width of the element. Defaults to `0`.
+
+*Inheritance:*
+
+[`SShapeElementImpl`](#sshapeelementimpl) &rarr; [`SChildElementImpl`](#schildelementimpl) &rarr; [`SParentElementImpl`](#sparentelementimpl) &rarr; [`SModelElementImpl`](#smodelelementimpl)
+
+### SShapeElementImpl
+
+Abstract class for elements with a position and size.
+
+*Properties:*
+
+* `position: Point`: The position of the element. Defaults to `{x: 0, y: 0}`.
+* `size: Dimension`: The size of the element. Defaults to `{width: -1, height: -1}`.
+* `layoutOptions: ModelLayoutOptions` - *optional*: Options for the layout of the element.
+
+*Inheritance:*
+
+[`SChildElementImpl`](#schildelementimpl) &rarr; [`SParentElementImpl`](#sparentelementimpl) &rarr; [`SModelElementImpl`](#smodelelementimpl)
+
+### SRoutableElementImpl
+
+Abstract class for edges.
+
+*Properties:*
+
+* `routerKind: string` - *optional*: The kind of router to use for the edge.
+* `routingPoints: Point[]` - *optional*: The routing points of the edge. Defaults to an empty array.
+* `sourceId: string`: The id of the source element.
+* `targetId: string`: The id of the target element.
+* `sourceAnchorCorrection: number` - *optional*: The correction of the source anchor. Defaults to `0`.
```

Comment:
what is a correction? a shift of the end/starting point of the edge in X/Y direction?